src/main.cpp

```diff
@@ -154,6 +154,29 @@ namespace {
      */
     map<uint256, NodeId> mapBlockSource;
 
+    /**
+     * Filter for transactions that were recently rejected by
+     * AcceptToMemoryPool. These are not rerequested until the chain tip
+     * changes, at which point the entire filter is cleared. Protected by
+     * cs_main.
+     *
+     * Without this filter we'd be re-requesting txs from each of our peers,
+     * increasing bandwidth consumption considerably. For instance, with 100
+     * peers, half of which relay a tx we don't accept, that might be a 50x
+     * bandwidth increase. A flooding attacker attempting to roll-over the
+     * filter using minimum-sized, 60byte, transactions might manage to send
+     * 1000/sec if we have fast peers, so we pick 120,000 to give our peers a
+     * two minute window to send invs to us.
+     *
+     * Decreasing the false positive rate is fairly cheap, so we pick one in a
+     * million to make it highly unlikely for users to have issues with this
+     * filter.
+     *
+     * Memory used: 1.7MB
+     */
+    CRollingBloomFilter recentRejects(120000, 0.000001, insecure_rand());
```

Comment:
Not sure if this should be insecure_rand. insecure_rand is meant for high-performance low-security use. 
- No performance constraints: this is done once, at initialization time
- If this value could be guessed by an adversary, there is additional DoS risk. 
- Also, the spread between nodes should be as uniform as possible to make sure that if false positives happen, they are at least as non-consistent as possible.
So IMO let's use the normal GetRand.

What I am also slightly worried about here is the static global initialization. Random generators will not have been initialized at that time.